ConcurrentAPI.adoc

```diff
@@ -0,0 +1,244 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature,
+provide concurrent and distributed programming capabilities.
+
+* Distributed Locks
+* Distributed CountDownLatch
+
+=== Scope
+
+All these interfaces should be placed in commons, in order to be used either in Embedded and Server mode.
+
+== Infinispan Distributed Locks
+
+=== DistributedLockManager and configuration
+
+```java
+package org.infinispan.lock.api;
+
+public class DistributedLockManager {
+   boolean defineLock(LockConfiguration configuration);
+   InfinispanLock get(String name);
```

Comment:
It seems to me that it is a big issue. Either you call `vertx.executeBlocking`(which you should avoid) or execute a blocking call in a thread that should not get blocked. Or you make sure that this `get(...)` method never blocks, by not doing any RPCs. Please choose.